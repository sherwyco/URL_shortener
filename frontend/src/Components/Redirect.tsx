import { useEffect } from 'react';
import { useParams, useNavigate } from 'react-router-dom';
import { getData } from '../Api/Client';
import { CircularProgress } from '@mui/material';

function Redirect() {
    const { shortCode } = useParams();
    const navigate = useNavigate()

    useEffect(() => {
        // eslint-disable-next-line
        getData(`/${shortCode}`, "", {}).then((response: any) => {
            window.location.replace(response.data.original_url)
        }).catch((error) => {
            console.log(error)
            navigate("/")
        })
    })
    return <><CircularProgress /></>
}

export default Redirect;