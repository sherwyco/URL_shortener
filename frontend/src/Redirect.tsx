import { useEffect } from 'react';
import { useParams, useNavigate } from 'react-router-dom';
import { getData } from './Api/Client';

function Redirect() {
    const { shortCode } = useParams();
    const navigate = useNavigate()

    useEffect(() => {
        // eslint-disable-next-line
        getData(`/${shortCode}`, "", {}).then((response: any) => {
            window.location.replace(response.data.original_url)
        }).catch((error) => {
            console.log(error)
            navigate("/")
        })
    })
    return <h1> You will be automatically redirected if link exists...</h1>
}

export default Redirect