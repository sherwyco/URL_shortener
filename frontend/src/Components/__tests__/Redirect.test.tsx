import renderer from 'react-test-renderer';
import Redirect from '../Redirect';

it('matches component snapshot', () => {
    const component = renderer.create(<Redirect />)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})