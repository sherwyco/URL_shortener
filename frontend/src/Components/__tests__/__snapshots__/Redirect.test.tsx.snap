// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component snapshot 1`] = `
<div
  style={
    {
      "display": "block",
    }
  }
>
  <div
    style={
      {
        "WebkitTransform": "translate(-50%, -50%)",
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      }
    }
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style={
        {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={{}}
        />
      </svg>
    </span>
  </div>
</div>
`;
